libs/fileset/glob.go

```diff
@@ -0,0 +1,45 @@
+package fileset
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+type GlobSet struct {
+	root     string
+	patterns []string
```

Comment:
We should consider making these patterns unix like, and calling `filepath.FromSlash` when we do glob matching. This approach would be similar to `gitignore` where all file paths use `/` as the path separator.

What I am not sure about with the current approach is whether glob patthers would work cross platform. Consider the case where a user develops on windows, and the CI runner runs linux. If they specify windows like glob patterns they would not work for the linux environment, but would work fine in their local windows env.

Reply:
Good point and +1 for the suggestion.